src/core/types/core_service.ts

```diff
@@ -19,7 +19,7 @@
 
 /** @internal */
 export interface CoreService<TSetup = void, TStart = void> {
-  setup(...params: any[]): Promise<TSetup>;
-  start(...params: any[]): Promise<TStart>;
-  stop(): Promise<void>;
+  setup(...params: any[]): TSetup | Promise<TSetup>;
+  start(...params: any[]): TStart | Promise<TStart>;
+  stop(): void | Promise<void>;
```

Comment:
I made this change to avoid making HttpService async and breaking a test utility that is used in quite a few places. I can also move this to a separate PR if we think that is better.

Reply:
I think it's alright and that it's actually closer to the various services implementations